pkg/cmd/cli/cmd/build.go

```diff
@@ -154,7 +154,7 @@ $ s2i build . centos/ruby-22-centos7 hello-world-app
 
 			d := docker.New(client, cfg.PullAuthentication)
 			err = d.CheckReachable()
-			if err != nil {
+			if len(cfg.AsDockerfile) == 0 && err != nil {
```

Comment:
you can wrap the d.CheckReachable() call with inside this if block, no?  No point in running it if we are going to ignore the result.